languages/swift/generic/Parse_swift_tree_sitter.ml

```diff
@@ -1849,13 +1849,18 @@ and map_lambda_literal (env : env) ((v1, v2, v3, v4) : CST.lambda_literal) :
     (* We inject into an `OtherStmtWithStmt` here because we needed somewhere for the
        capture list to go. This is not permitted with just a regular statement, so there is
        a new variant `OSWS_Closure` to signify this.
+       We put it into the top-level statements so that we do not prevent the inner statements
+       from surviving IL translation. It shouldn't mess anything up, because it shouldn't be
+       translated.
     *)
+    let capture_group_stmt =
+      G.exprstmt
+        (G.OtherExpr
+           (("CaptureGroup", v1), Common.map (fun x -> G.Pa x) captures)
+        |> G.e)
+    in
     G.FBStmt
-      (G.OtherStmtWithStmt
```

Comment:
I'm not arguing for one solution or the other, but is this better than adding a case for `OSWS_Closure` in `AST_to_IL` ? Why was `OSWS_Closure` added in the first place?

Reply:
it was added because at that time, i didn't know enough to use `OtherStmt` as much as ive learned to now. i really wanted somewhere for the capture list to go, and i wanted it to be "nice", but now i understand that this suffices

i also now reviewed what capture lists in Swift do and i am highly suspicious we would ever use it for semantic analysis. so now i am fine throwing it in the garbage. before, i think i had hopes we might use it.